### DISCUSSION TITLE
mobx6: Organizing code in real app with multiple stores

### DISCUSSION THREAD

**Discussion by mogadanez** [mobxjs/mobx#2657] [Q&A]
entry point:
https://mobx.js.org/react-integration.html#using-external-state-in-observer-components
using React context example
```
import {observer} from 'mobx-react-lite'
import {createContext, useContext} from "react"

const TimerContext = createContext<Timer>()

const TimerView = observer(() => {
    // Grab the timer from the context.
    const timer = useContext(TimerContext) // See the Timer definition above.
    return (
        <span>Seconds passed: {timer.secondsPassed}</span>
    )
})

ReactDOM.render(
    <TimerContext.Provider value={new Timer()}
        <TimerView />
    </TimerContext.Provider>,
    document.body
)
``` 
in this example  all in one place and works fine

but if  extract  component to separate file:
```
const TimerView = observer(() => {
    // Grab the timer from the context.
    const timer = useContext(TimerContext) // See the Timer definition above.
    return (
        <span>Seconds passed: {timer.secondsPassed}</span>
    )
})

```
how I can access  TimerContext?

Also, how to make it for multiple stores?
wrapping   as
```<TimerContext.Provider value={new Timer()}
...    
</TimerContext.Provider>
``` 
for each one  not  very fancy

**Comment by danielkcz** [comment]
> how I can access TimerContext?

Uh, you export it? :)

> Also, how to make it for multiple stores?

Wrap it into an object?

**Comment by akaguny** [comment]
maybe usefull http://mobx-easy.georgy-glezer.com/docs/getting-root/ for dependencie injection(DI) without "constructor drilling", also you may create subtrees of your store for specific pages (you make new context, that included only stores, what you need for page)
i think that main pattern is trees of stores, and if you use helpers for DI.
If you want i can open my with POC for organize app, i can not take it public for ALL because it for study and POC =)
